File: languages/python/bloock/entity/identity/verification_receipt.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto


class VerificationReceipt:
    """
    Represents a receipt for a verification session.
    """
    def __init__(self, session_id: int, verification_request: str) -> None:
        """
        Constructs a VerificationReceipt object with the specified parameters.
        :type verification_request: object
        :type session_id: object
        :rtype: object
        """
        self.session_id = session_id
        self.verification_request = verification_request

    @staticmethod
    def from_proto(c: proto.VerificationReceipt) -> VerificationReceipt:
        return VerificationReceipt(
            session_id=c.session_id,
            verification_request=c.verification_request,
        )

    def to_proto(self) -> proto.VerificationReceipt:
        return proto.VerificationReceipt(
            session_id=self.session_id,
            verification_request=self.verification_request,
        )

File: languages/php/Bloock/Entity/Identity/VerificationReceipt.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents a receipt for a verification session.
 */
class VerificationReceipt
{
    private int $sessionID;
    private string $verificationRequest;

    /**
     * Constructs a VerificationReceipt object with the specified parameters.
     * @param int $sessionID
     * @param string $verificationRequest
     */
    public function __construct(int $sessionID, string $verificationRequest)
    {
        $this->sessionID = $sessionID;
        $this->verificationRequest = $verificationRequest;
    }

    public static function fromProto(\Bloock\VerificationReceipt $res): VerificationReceipt
    {
        return new VerificationReceipt($res->getSessionId(), $res->getVerificationRequest());
    }

    /**
     * Gets the session id of the verification.
     * @return int
     */
    public function getSessionID(): int
    {
        return $this->sessionID;
    }

    /**
     * Gets the verification request json.
     * @return string
     */
    public function getVerificationRequest(): string
    {
        return $this->verificationRequest;
    }

    public function toProto(): \Bloock\VerificationReceipt
    {
        $p = new \Bloock\VerificationReceipt();
        $p->setSessionId($this->sessionID);
        $p->setVerificationRequest($this->verificationRequest);
        return $p;
    }
}
